src/Microsoft.AspNetCore.Http.Extensions/RangeParser.cs

```diff
@@ -0,0 +1,83 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using Microsoft.AspNetCore.Http.Headers;
+using Microsoft.Extensions.Primitives;
+using Microsoft.Net.Http.Headers;
+
+namespace Microsoft.AspNetCore.Http.Extensions
+{
+    /// <summary>
+    /// Provides a parser for the Range Header in an <see cref="HttpContext.Request"/>.
+    /// </summary>
+    public class RangeParser
+    {
+        /// <summary>
+        /// Returns the requested range if the Range Header in the <see cref="HttpContext.Request"/> is valid.
+        /// </summary>
+        /// <param name="context">The <see cref="HttpContext"/> associated with the request.</param>
+        /// <param name="requestHeaders">The <see cref="RequestHeaders"/> associated with the given <paramref name="context"/>.</param>
+        /// <param name="lastModified">The <see cref="DateTimeOffset"/> representation of the last modified date of the file.</param>
+        /// <param name="etag">The <see cref="EntityTagHeaderValue"/> provided in the <see cref="HttpContext.Request"/>.</param>
+        /// <returns>A collection of <see cref="RangeItemHeaderValue"/> containing the ranges parsed from the <paramref name="requestHeaders"/>.</returns>
+        public static ICollection<RangeItemHeaderValue> ParseRange(HttpContext context, RequestHeaders requestHeaders, DateTimeOffset lastModified, EntityTagHeaderValue etag)
```

Comment:
Some of your MVC file result types won't always have an etag or lastmodified right? What will you do there? or will null work as expected?